runironic.sh

```diff
@@ -1,53 +1,53 @@
-    crudini --set /etc/my.conf mysqld innodb_log_buffer_size 512K
-    mysql_install_db --datadir=$DATADIR
-    chown -R mysql /var/log/mariadb
-    chown -R mysql $DATADIR
-    cd /usr
-    mysqld_safe --datadir=$DATADIR --user mysql &
-    sleep 1
-    ironic_password=$(echo $(date;hostname)|sha256sum |cut -c-20)
-    mysqladmin -u root password $ironic_password
-    cat > /tmp/configure-mysql.sql << EOF
-TRUNCATE mysql.user;
-CREATE USER 'ironic'@'localhost' identified by '$ironic_password';
-GRANT ALL on *.* TO 'ironic'@'localhost' WITH GRANT OPTION;
-DROP DATABASE IF EXISTS test;
-CREATE DATABASE ironic;
-FLUSH PRIVILEGES;
-EOF
-    mysql -u root -p$ironic_password -h 127.0.0.1 < /tmp/configure-mysql.sql
-    ps auxf
-    crudini --set /etc/ironic/ironic.conf database connection mysql+pymysql://ironic:$ironic_password@localhost/ironic?charset=utf8
-    ironic-dbsync --config-file /etc/ironic/ironic.conf create_schema
-    touch $DATADIR/ironic-initalized
-else
-    mysqld_safe --datadir=$DATADIR --user mysql &
-fi
-
 # Get environment settings and update ironic.conf
 IP=${IP:-"172.22.0.1"}
 HTTP_PORT=${HTTP_PORT:-"80"}
 INTERFACE=${INTERFACE:-"provisioning"}
-
-sed -i -e s/IRONIC_IP/$IP/g -e s/HTTP_PORT/$HTTP_PORT/g /etc/ironic/ironic.conf 
+MARIADB_PASSWORD=${MARIADB_PASSWORD:-"change_me"}
 
 # Allow access to Ironic
 if ! iptables -C INPUT -i $INTERFACE -p tcp -m tcp --dport 5050 -j ACCEPT > /dev/null 2>&1; then
     iptables -I INPUT -i $INTERFACE -p tcp -m tcp --dport 5050 -j ACCEPT
 fi
 
-/usr/bin/python2 /usr/bin/ironic-conductor > /var/log/ironic-conductor.out 2>&1 &
+cp /etc/ironic/ironic.conf /etc/ironic/ironic.conf_orig
+
+crudini --set /etc/ironic/ironic.conf DEFAULT auth_strategy noauth
+crudini --set /etc/ironic/ironic.conf DEFAULT my_ip ${IP}
+crudini --set /etc/ironic/ironic.conf DEFAULT debug true
+crudini --set /etc/ironic/ironic.conf DEFAULT default_network_interface noop
+crudini --set /etc/ironic/ironic.conf DEFAULT enabled_boot_interfaces pxe,ipxe
+crudini --set /etc/ironic/ironic.conf DEFAULT enabled_power_interfaces ipmitool,idrac
+crudini --set /etc/ironic/ironic.conf DEFAULT enabled_management_interfaces ipmitool,idrac
+crudini --set /etc/ironic/ironic.conf DEFAULT enabled_hardware_types ipmi,idrac
+crudini --set /etc/ironic/ironic.conf DEFAULT enabled_vendor_interfaces ipmitool,no-vendor,idrac
+crudini --set /etc/ironic/ironic.conf DEFAULT default_boot_interface ipxe
+crudini --set /etc/ironic/ironic.conf DEFAULT default_deploy_interface direct
+crudini --set /etc/ironic/ironic.conf DEFAULT enabled_inspect_interfaces inspector,idrac
+crudini --set /etc/ironic/ironic.conf DEFAULT default_inspect_interface inspector
+crudini --set /etc/ironic/ironic.conf DEFAULT rpc_transport json-rpc
+crudini --set /etc/ironic/ironic.conf dhcp dhcp_provider none
+crudini --set /etc/ironic/ironic.conf conductor automated_clean false
+crudini --set /etc/ironic/ironic.conf conductor api_url http://${IP}:${HTTP_PORT}
```

Comment:
I think a couple typos in transition from Dockerfile
api_url should be http://${IP}:6385

Reply:
copy/paste is just too hard for me :/
fixed

Reply:
Thanks, looks good!